cpp/ql/test/library-tests/dataflow/dataflow-tests/IRDataflowTestCommon.qll

```diff
@@ -36,9 +36,27 @@ class TestAllocationConfig extends DataFlow::Configuration {
     )
   }
 
+  override predicate isAdditionalFlowStep(DataFlow::Node n1, DataFlow::Node n2) {
+    exists(GlobalOrNamespaceVariable var | var.getName().matches("flowTestGlobal%") |
+      writesVariable(n1.asInstruction(), var) and
+      var = n2.asVariable()
+      or
+      readsVariable(n2.asInstruction(), var) and
+      var = n1.asVariable()
+    )
+  }
+
   override predicate isBarrier(DataFlow::Node barrier) {
     barrier.asExpr().(VariableAccess).getTarget().hasName("barrier")
   }
 
   override predicate isBarrierGuard(DataFlow::BarrierGuard bg) { bg instanceof TestBarrierGuard }
 }
+
+private predicate readsVariable(LoadInstruction load, Variable var) {
```

Comment:
It's a shame we have to duplicate code (from DefaultTaintTracking.qll) here.

Reply:
The duplication might look less bad if we just in-line it.

Reply:
Yes, and it isn't a lot of code (I thought we had a mechanism for testing things that are `private`?)

Reply:
Quick Evaluation can now peek through private items, but qltest can't. The standard workaround is to make the private items public in a public `module Internal`, and then users know they're asking for trouble if they access anything in that module.

I can also delete the change to the tests in this directory. I wrote them for my own debugging purposes, but they are probably redundant now that the code is working, and there are better tests in the `DefaultTaintTracking` dir.

Reply:
I'm happy with inlining it.